src/app/year-calendar/year-calendar.component.ts

```diff
@@ -0,0 +1,254 @@
+          this.componentConfig.min
+        );
+    this.inputValueType = this.utilsService.getInputType(this.inputValue, this.componentConfig.allowMultiSelect);
+    this._shouldShowCurrent = this.shouldShowCurrent();
+  }
+
+  writeValue(value: CalendarValue): void {
+    this.inputValue = value;
+
+    if (value) {
+      this.selected = this.utilsService
+        .convertToMomentArray(value, this.componentConfig.format, this.componentConfig.allowMultiSelect);
+      this.yearMatrix = this.yearCalendarService
+        .generateYear(this.componentConfig, this.currentDateView, this.selected);
+      this.inputValueType = this.utilsService.getInputType(this.inputValue, this.componentConfig.allowMultiSelect);
+    }
+  }
+
+  registerOnChange(fn: any): void {
+    this.onChangeCallback = fn;
+  }
+
+  onChangeCallback(_: any) {
+  };
+
+  registerOnTouched(fn: any): void {
+  }
+
+  validate(formControl: FormControl): ValidationErrors | any {
+    if (this.minDate || this.maxDate) {
+      return this.validateFn(formControl.value);
+    } else {
+      return () => null;
+    }
+  }
+
+  processOnChangeCallback(value: Moment[]): CalendarValue {
+    return this.utilsService.convertFromMomentArray(
+      this.componentConfig.format,
+      value,
+      this.componentConfig.returnedValueType || this.inputValueType
+    );
+  }
+
+  initValidators() {
+    this.validateFn = this.validateFn = this.utilsService.createValidator(
+      {minDate: this.minDate, maxDate: this.maxDate},
+      this.componentConfig.format,
+      'year');
+
+    this.onChangeCallback(this.processOnChangeCallback(this.selected));
+  }
+
+  yearClicked(year: IYear) {
+    this.selected = this.utilsService
+      .updateSelected(this.componentConfig.allowMultiSelect, this.selected, year, 'year');
+    this.yearMatrix = this.yearCalendarService
+      .generateYear(this.componentConfig, this.currentDateView, this.selected);
+    this.onSelect.emit(year);
+  }
+
+  onLeftNav() {
+    this.currentDateView = this.currentDateView.clone().subtract(12, 'year');
+    this.yearMatrix = this.yearCalendarService.generateYear(this.componentConfig, this.currentDateView, this.selected);
+  }
+
+  onLeftSecondaryNav() {
+    let navigateBy = this.componentConfig.multipleYearsNavigateBy;
+    const isOutsideRange = this.componentConfig.min &&
+      this.currentDateView.year() - this.componentConfig.min.year() < navigateBy;
+
+    if (isOutsideRange) {
+      navigateBy = this.currentDateView.year() - this.componentConfig.min.year();
+    }
+    this.currentDateView = this.currentDateView.clone().subtract(navigateBy, 'year');
+    this.yearMatrix = this.yearCalendarService.generateYear(this.componentConfig, this.currentDateView, this.selected);  // remove?
+  }
+
+  onRightNav() {
+    this.currentDateView = this.currentDateView.clone().add(12, 'year');
+    this.yearMatrix = this.yearCalendarService.generateYear(this.componentConfig, this.currentDateView, this.selected);  // remove?
```

Comment:
remove???

Reply:
As above -- I've created the year calendar from the design of the month calendar. In `month-calendar.component.ts`, there is a, perhaps, strange difference between methods `onLeftNav()` and `onRightNav()`; only the former actually updates the `yearMonths` object. Why doesn't the "right" handler also update the object? I opted to update `yearMatrix` in both methods, but marked the code to draw your attention.

Reply:
Any thoughts on this? :)

Reply:
this is very odd... I need to check. Sorry for the long responses...